### PR TITLE
Add event batching on the server

### LINKED ISSUE #7: Batch events emitted from `EventEmitter.of(ctx)`
It is useful to batch events emitted on the server to
allow plugins to add mappers that are dependent on data 
coming from middleware.

### PR DESCRIPTION
**Pull request by ganemone**
Fixes #7

### CODE REVIEW

**Comment by lhorie** [comment]
Need docs

**Comment by nadiia** [comment]
Please update the readme

File: README.md [hunk 1]
```diff
@@ -114,7 +133,7 @@ Registers a callback to be called when an event of a type is emitted. Note that
 #### `events.emit`
 
 ```js
-events.emit(type, payload, ...args)
+events.emit(type, payload)
```
Comment (nadiia) [inline README.md]: seems like the `ctx` should go as an optional arg here? line 141 also needs to be updated

Comment (ganemone) [reply README.md]: I actually don't allow emitting events with ctx as a third argument. It is added by using `EventEmitter.of(ctx)`

**Comment by ganemone** [comment]
!merge

File: src/server.js [hunk 2]
```diff
@@ -21,37 +21,61 @@
...
         if (items) {
-          items.forEach(({type, payload}) => {
-            this.of().emit(type, payload, ctx);
-          });
-          await next();
+          for (let index = 0; index < items.length; index++) {
+            const {type, payload} = items[index];
+            emitter.emit(type, payload);
+          }
           ctx.status = 200;
```
Comment (AlexMSmithCA) [inline src/server.js]: Does it matter that we are now updating status downstream here?

Comment (ganemone) [reply src/server.js]: yea, I think we should set it as early as possible

**Pull request merged**